ipatests/test_integration/test_sssd.py

```diff
@@ -403,3 +405,141 @@ def test_trustdomain_disable_disables_subdomain(self):
+        """
+        master = self.master
+        client = self.clients[0]
+        tasks.kinit_admin(master)
+
+        # add hbacrule
+        master.run_command(['ipa', 'hbacrule-add', '--usercat', 'all',
+                            '--hostcat', 'all', '--servicecat', 'all',
+                            'any_to_any'])
+        # add a user
+        username = "testuser" + str(random.randint(200000, 9999999))
+        userpasswd = 'Secret123'
+
+        input_passwd = '{upasswd}\n{upasswd}\n'.format(upasswd=userpasswd)
+        tasks.user_add(master, username, password=userpasswd)
+
+        cmd_output = master.run_command(['ipa', 'passwd', username],
+                                        stdin_text=input_passwd)
+        assert 'Changed password' in cmd_output.stdout_text
+
+        # add group a
+        cmd_output = master.run_command(['ipa', 'group-add', 'a'])
+        assert 'Added group "a"' in cmd_output.stdout_text
+
+        # add group b
+        cmd_output = master.run_command(['ipa', 'group-add', 'b'])
+        assert 'Added group "b"' in cmd_output.stdout_text
+
+        # add group c
+        cmd_output = master.run_command(['ipa', 'group-add', 'c'])
+        assert 'Added group "c"' in cmd_output.stdout_text
+
+        # add group members
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'b', '--groups', 'a'])
+        assert 'Group name: b' in cmd_output.stdout_text
+        assert 'Member groups: a' in cmd_output.stdout_text
+        assert 'Number of members added 1' in cmd_output.stdout_text
+
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'c', '--groups', 'b'])
+        assert 'Group name: c' in cmd_output.stdout_text
+        assert 'Member groups: b' in cmd_output.stdout_text
+        assert 'Indirect Member groups: a' in cmd_output.stdout_text
+
+        # add user to group 'a'
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'a', '--users', username])
+        assert 'Group name: a' in cmd_output.stdout_text
+        assert 'Member users: {}'.format(username) in cmd_output.stdout_text
+        assert 'Member of groups: b' in cmd_output.stdout_text
+        assert 'Indirect Member of group: c' in cmd_output.stdout_text
+
+        # clear sssd_cache
+        clear_sssd_cache(master)
+
+        # user lookup
+        cmd = 'ipa user-show {} | grep group'.format(username)
+        cmd_output = master.run_command(cmd)
+        assert_str = 'Member of groups: a, ipausers'
+        assert assert_str in cmd_output.stdout_text
+        assert_str = 'Indirect Member of group: b, c'
+        assert assert_str in cmd_output.stdout_text
+
+        clear_sssd_cache(master)
+
+        # ssh to client using the user created above from master
+        input_passwd = '{upasswd}\n{upasswd}\n{upasswd}\n{upasswd}\n'\
```

Comment:
We already changes password for the user. Do we need to specify password 4 times again?

Reply:
When a new user is created and you try to use that user for ssh,
it expires the password and prompts you for a new one

Reply:
fixed it